Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/AWSS3ErrorWith200StatusXMLMiddleware.swift

```diff
@@ -40,9 +35,41 @@ extension AWSS3ErrorWith200StatusXMLMiddleware: HttpInterceptor {
         context: some MutableResponse<Self.InputType, Self.RequestType, Self.ResponseType>
     ) async throws {
         let response = context.getResponse()
-        if try await isErrorWith200Status(response: response) {
-            response.statusCode = errorStatusCode
-            context.updateResponse(updated: response)
+
+        // Check if the status code is OK (200)
+        guard response.statusCode == .ok else {
+            return
+        }
+
+        guard let data = try await response.body.readData() else {
+            return
+        }
+
+        let statusCode = try isRootErrorElement(data: data) ? errorStatusCode : response.statusCode
```

Comment:
How do we know we will have enough data here to determine if the body returns an error?
`readData()` could return zero bytes if the data is not yet available, and the stream could be filled with error data after this step is complete.

Reply:
we don't get the stream back all at once, each response is a separate action so it should be checking each streamed message for <Error> at root